crates/steel-core/src/primitives/bytevectors.rs

```diff
@@ -320,3 +307,51 @@ pub fn bytes_append(value: &SteelByteVector, other: &SteelByteVector) -> Result<
             .collect(),
     )))
 }
+
+/// Decodes a string from a bytevector containing valid UTF-8.
+///
+/// (utf8->string buf [start] [end]) -> string?
+///
+/// * buf : bytes?
+/// * start: int? = 0
+/// * end: int? = (bytes-length buf)
+///
+/// # Examples
+/// ```scheme
+/// (utf8->string (bytes #xe5 #x8d #x83 #xe8 #x91 #x89)) ;; => "千葉"
+/// ```
+#[function(name = "utf8->string")]
```

Comment:
:question: Should we call this `bytes->string` instead?

Reply:
I think the r7rs spec uses `utf8->string`, but racket uses `bytes->string/utf8` (see [here](https://docs.racket-lang.org/reference/bytestrings.html#%28def._%28%28quote._~23~25kernel%29._bytes-~3estring%2Futf-8%29%29) - maybe we just include both?